sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveScriptTransformationSuite.scala

```diff
@@ -528,4 +530,65 @@ class HiveScriptTransformationSuite extends BaseScriptTransformationSuite with T
     checkAnswer(query2, identity, Row("\\N,\\N,\\N") :: Nil)
 
   }
+
+  test("SPARK-34879: HiveInspector supports DayTimeIntervalType and YearMonthIntervalType") {
+    assume(TestUtils.testCommandAvailable("/bin/bash"))
+    withTempView("v") {
+      val df = Seq(
+        (Duration.ofDays(1),
+          Duration.ofSeconds(100).plusNanos(123456),
+          Duration.ofSeconds(Long.MaxValue / DateTimeConstants.MICROS_PER_SECOND),
+          Period.ofMonths(10)),
+        (Duration.ofDays(1),
+          Duration.ofSeconds(100).plusNanos(1123456789),
+          Duration.ofSeconds(Long.MaxValue / DateTimeConstants.MICROS_PER_SECOND),
+          Period.ofMonths(10))
+      ).toDF("a", "b", "c", "d")
+        .select('a, 'b, 'c.cast(DayTimeIntervalType).as("c_1"), 'd)
```

Comment:
`'a` is deprecated in the future scala versions. I wouldn't use it.

Reply:
> `'a` is deprecated in the future scala versions. I wouldn't use it. Let's replace it by `$"a"`

Done